projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -277,17 +297,42 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      */
     public show(message?: string): void {
         clearInterval(this.timeoutId);
-        if (message !== undefined) { this.toastMessage = message; }
+
+        const overlaySettings: OverlaySettings = {
+            positionStrategy: new GlobalPositionStrategy({
+                horizontalDirection: HorizontalAlignment.Center,
+                verticalDirection:
+                    this.position === 'bottom'
+                        ? VerticalAlignment.Bottom
+                        : this.position === 'middle'
+                        ? VerticalAlignment.Middle
+                        : VerticalAlignment.Top,
+            }),
+            closeOnEscape: false,
+            closeOnOutsideClick: false,
+            modal: false,
+            outlet: this.outlet
+        };
+
+        if (message !== undefined) {
+            this.toastMessage = message;
+        }
+
+        this.overlayId = this.overlayService.attach(
+            this.elementRef,
+            overlaySettings
+        );
+
         this.onShowing.emit(this);
-        this.isVisible = true;
-        this.animationState = 'visible';
+        this._isVisible = true;
```

Comment:
because of the animation toast is not actually 100% visible here. You could handle `onOpened` event of the `IgxOverlayService` and set _isVisible there.

Reply:
Same here. The `isVisible` property is updated in the `onOpening` event of the toggle directive.